pkg/storage/watch_cache_test.go

```diff
@@ -230,6 +231,25 @@ func TestEvents(t *testing.T) {
 	}
 }
 
+func TestWaitUntilFreshAndList(t *testing.T) {
+	store := newWatchCache(3)
+
+	// In background, wait some time and then update the store.
```

Comment:
It's probably not guaranteed, but there is very very high probability (in fact I run this test 100 times and it was always the case), the WaitUtilFresh will be called before any pod will be added - and this is the case that I would like to test there.
Without this sleep - it happens that that those pods are already there before WaitUntil... is called.

But if you think that it's not worth having it, I'm fine with removing it. WDYT?